instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/AdviceScope.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.play.v2_4;
+
+import static io.opentelemetry.javaagent.instrumentation.play.v2_4.Play24Singletons.instrumenter;
+import static io.opentelemetry.javaagent.instrumentation.play.v2_4.Play24Singletons.updateSpan;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import javax.annotation.Nullable;
+import play.api.mvc.Action;
+import play.api.mvc.Request;
+import play.api.mvc.Result;
+import scala.concurrent.Future;
+
+/** Container used to carry state between enter and exit advices */
+public final class AdviceScope {
```

Comment:
isn't the usual pattern to have the `AdviceScope` class as a nested class in the advice class?

Reply:
Yes, unless there are numerous instrumentation advices that define the same, which is not the case here.